prog/kubernetes/provision_kubernetes_node.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+class Prog::Kubernetes::ProvisionKubernetesNode < Prog::Base
+  subject_is :kubernetes_cluster
+
+  def vm
+    @vm ||= Vm[frame.fetch("vm_id", nil)] || nil
+  end
+
+  def write_hosts_file_if_needed(ip = nil)
+    return unless Config.development?
+    return if vm.sshable.cmd("sudo cat /etc/hosts").match?(/#{kubernetes_cluster.endpoint}/)
+    ip = kubernetes_cluster.vms.first.ephemeral_net4 if ip.nil?
+
+    vm.sshable.cmd("echo '#{ip} #{kubernetes_cluster.endpoint}' | sudo tee -a /etc/hosts")
+  end
+
+  label def start
+    name = "#{kubernetes_cluster.name.downcase}-control-plane-#{SecureRandom.alphanumeric(5).downcase}"
+
+    vm_st = Prog::Vm::Nexus.assemble_with_sshable(
+      "ubi",
+      kubernetes_cluster.projects.first.id,
+      # we should reiterate how we name the vm. some how correlate it to the vm's info.
```

Comment:
Generally, we use the related object's UBID. For example, the name of the runner VM is the runner's UBID, and the name of the Postgres server's VM is its UBID

Reply:
We might use ubid + random key instead of name + random

Reply:
I'm inclined towards postponing this decision to post-alpha as we have a few decisions to make
1. What will the Clover names of CP VMs be?
2. What about the worker VMs?
3. We also have cluster names of the nodes. Currently it's the VM inhost name.